consensus/vbft/block_pool.go

```diff
@@ -623,7 +616,18 @@ func (pool *BlockPool) setBlockSealed(block *Block, forEmpty bool) error {
 	if !forEmpty {
 		c.SealedBlock = block
 	} else {
-		block.Block.Transactions = nil // remove its payload
+		// FIXME: check if missed any other system-txns
+		systemTxs := make([]*types.Transaction, 0)
+		emptyAddr := common.Address{}
+		for _, tx := range block.Block.Transactions {
+			if tx.TxType == types.Invoke && bytes.Compare(tx.Payer[:], emptyAddr[:]) == 0 {
+				systemTxs = append(systemTxs, tx)
+			} else {
+				break
+			}
+		}
+
+		block.Block.Transactions = systemTxs
```

Comment:
when dealing with `block, transaction` etc, we implicitly assume they are immutable after construction. here you have modified block's transactions, so you need explicitly recalculate transaction root and set block.Hash to nil.